### PR TITLE
[go_router] add go_router example with bloc

### PR DESCRIPTION
**Pull request by Junddao**
- add authentication bloc for manage login status
- use redirect and refreshListable in go_router

image
image

*Replace this paragraph with a description of what this PR is changing or adding, and why. Consider including before/after screenshots.*

*List which issues are fixed by this PR. You must list at least one issue.*

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [] I updated/added relevant documentation (doc comments with `///`).
- [] I added new tests to check the change I am making, or this PR is [test-exempt].
- [] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/go_router/example/lib/auth_bloc/bloc/authentication_bloc.dart [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+// found in the LICENSE file.
+
+import 'package:equatable/equatable.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import '../type/authentication_status_type.dart';
+
+part 'authentication_event.dart';
+part 'authentication_state.dart';
+
+/// A [Bloc] which manages users authentication .
```
Comment (Hixie) [inline packages/go_router/example/lib/auth_bloc/bloc/authentication_bloc.dart:12]: trivial nit: you have an extra space before the period

File: packages/go_router/example/lib/auth_bloc/bloc/authentication_event.dart [hunk 2]
```diff
@@ -0,0 +1,29 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+part of 'authentication_bloc.dart';
```
Comment (Hixie) [inline packages/go_router/example/lib/auth_bloc/bloc/authentication_event.dart:5]: nit: our style guide discourages the use of `part of`.

Comment (Hixie) [reply packages/go_router/example/lib/auth_bloc/bloc/authentication_event.dart:5]: (https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-using-library-and-part-of)

Comment (Junddao) [reply packages/go_router/example/lib/auth_bloc/bloc/authentication_event.dart]: solved. please check again.

File: packages/go_router/example/lib/auth_bloc/bloc/authentication_bloc.dart (hunk 1 above)
Comment (Junddao) [reply packages/go_router/example/lib/auth_bloc/bloc/authentication_bloc.dart:10]: solved. please check again.

**Comment by chunhtai** [comment]
Hi @Junddao, I think this may not be the best place to put this example. We should keep the example in flutter/package not depending on third-party package. The examples are supposed to be focusing on go_router only. Would you consider putting this example in the flutter_bloc instead?

I noticed the example also uses provider, I think we should remove it as well.

**Comment by Junddao** [comment]
> Hi @Junddao, I think this may not be the best place to put this example. We should keep the example in flutter/package not depending on third-party package. The examples are supposed to be focusing on go_router only. Would you consider putting this example in the flutter_bloc instead?
> 
> I noticed the example also uses provider, I think we should remove it as well.

ok.  Thank you for your reply.   
I will add this example  in flutter_bloc package.

🙏

**Pull request closed without merging**